db/design_doc.go

```diff
@@ -244,7 +244,7 @@ func filterViewResult(input sgbucket.ViewResult, user auth.User, applyChannelFil
 			value, ok := row.Value.([]interface{})
 			if ok {
 				// value[0] is the array of channels; value[1] is the actual value
-				if !hasStarChannel || channelsIntersect(visibleChannels, value[0].([]interface{})) {
+				if !hasStarChannel || channelsIntersect(visibleChannels, value[0].([]string)) {
```

Comment:
How is this related to otto?

Reply:
If the underlying issue is the way walrus uses otto for map function execution, should this be fixed there?

Reply:
The unit tests are using wulrus, in walrus views.go in _compileDesignDoc it calls back on:

sgbucket.NewJSMapFunction

Which internally is using otto.

I'm trying to understand if this is strictly a walrus issue or an SG issue.

Reply:
Looking at Traun's comments below for integration test, then it looks like this should be treated as a walrus issue. I'll revert the PR code change and rerun the integration tests.